src/helper/fetch-manifest.js

```diff
@@ -47,6 +47,15 @@ module.exports = function (scope) {
                                   return Promise.resolve(response);
                                 default:
                                   options.token = options.data.token || scope.token;
+                                  // add allow cache headers
+                                  if(scope.cache && scope.cache[path[0]] && scope.cache[path[0]][key]) {
```

Comment:
What about differing levels of iteration?
The `user.me` function might be on a different level than some message functions will be.

Reply:
Can you give me an example?
Most of the time there is not really a valid case for using deeply nested REST resources, and it leads to problematic and (ugly) code, trying to destructure the path into resource and params.

e.g. 
PUT user/1/message/2/archive
Should be either:
  PUT  message/archive   { message: 2 }   
  PUT message/archive/?message=2

Reply:
A good example would be the following already-implemented routes, which are all on different levels.

```
/api/versions
/api/v1/manifest.json
/api/v1/user/me.json
```

Reply:
path[0] = user :)

Reply:
The code in this file particularly is lacking clarity due to poor naming, docblocks, comments etc.
path[0]  is in fact the api ref, minus the basepath "api/<version> ".
I'm currently doing some refactoring work to tidy up some code, so this will be addressed in the future 
I will just add a test for now to keep it in check.

Reply:
@finwo suggesting to close this for now and move on, and revisit this if the need arises to cache more deeply nested apis in the future. 
Writing tests for js-api-client is really out of scope for this issue too, because a fake backend needs to be made etc.

Reply:
So, close & write it when it's needed then?